cmake/modules/CompilerFlags.cmake

```diff
@@ -29,12 +32,13 @@ if(NOT MSVC)
 	endif()
 
 	if(BUILD_WARNINGS_AS_ERRORS)
+		set(CMAKE_COMPILE_WARNING_AS_ERROR ON)
 		set(CMAKE_SUPPRESSED_WARNINGS "-Wno-unused-parameter -Wno-missing-field-initializers -Wno-sign-compare -Wno-type-limits -Wno-implicit-fallthrough -Wno-format-truncation")
-		set(FALCOSECURITY_LIBS_COMMON_FLAGS "${FALCOSECURITY_LIBS_COMMON_FLAGS} -Wextra -Werror ${CMAKE_SUPPRESSED_WARNINGS}")
+		set(FALCOSECURITY_LIBS_COMMON_FLAGS "${FALCOSECURITY_LIBS_COMMON_FLAGS} -Wextra ${CMAKE_SUPPRESSED_WARNINGS}")
 	endif()
 
 	set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${FALCOSECURITY_LIBS_COMMON_FLAGS}")
-	set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${FALCOSECURITY_LIBS_COMMON_FLAGS} -std=c++0x")
+	set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${FALCOSECURITY_LIBS_COMMON_FLAGS} -std=c++17")
```

Comment:
Using `set(CMAKE_CXX_STANDARD 17)`  with  Cmake `3.22.1` is not enough, because cmake see `gnu17` as default and it doesn't override it with `c++17` this causes the following error we have seen in CI:

```
/tmp/libs/build/libsinsp/cri.pb.h:3359:53: error: expected unqualified-id before numeric constant
 3359 |   const ::runtime::v1alpha2::LinuxPodSandboxConfig& linux() const;
```

You can notice the difference with the following commands:

```bash
g++ -dM -E -x c /dev/null | grep linux

#define __linux 1
#define __gnu_linux__ 1
#define linux 1
#define __linux__ 1
```
with `gnu` we have the macro `linux` defined

```bash
g++ -std=c17 -dM -E -x c /dev/null | grep linux

#define __linux 1
#define __gnu_linux__ 1
#define __linux__ 1
```
with `c++17` we don't have it! The CI was failing because even if we set `set(CMAKE_CXX_STANDARD 17)`, CMake doesn't set `c++17` an preserves `gnu17`

This seems to be a CMake misbehavior https://gitlab.kitware.com/cmake/cmake/-/issues/22224 that is solved in versions >= `3.22.1`. We tested it with cmake `3.25.2`

Thank you @FedeDP for the huge help!